src/gfx/gizmos/scaling_gizmo.hpp

```diff
@@ -0,0 +1,228 @@
+      fwd[i] = 1.0f;
+      float3 t1 = float3(-fwd.z, -fwd.x, fwd.y);
+      float3 t2 = linalg::cross(fwd, t1);
+
+      // Slightly decrease hitbox size if view direction is parallel
+      // e.g. looking towards +z you are less likely to want to click on the z axis
+      float dotThreshold = 0.8f;
+      float angleFactor = std::max(0.0f, (linalg::abs(linalg::dot(localRayDir, fwd)) - dotThreshold) / (1.0f - dotThreshold));
+
+      // Make hitboxes slightly bigger than the actual visuals
+      const float2 hitboxScale = linalg::lerp(float2(2.2f, 1.2f), float2(0.8f, 1.0f), angleFactor);
+
+      auto &min = handle.selectionBox.min;
+      auto &max = handle.selectionBox.max;
+      if (i == 3) {
+        float hitbox_size = getGlobalAxisLength() * 0.2f;
+        min = float3(-hitbox_size, -hitbox_size, -hitbox_size);
+        max = float3(hitbox_size, hitbox_size, hitbox_size);
+      } else {
+        // The size of the selection box for the axis handles is reduced to avoid overlapping with the hitbox of the centered cube.
+        min = (-t1 * getGlobalAxisRadius() - t2 * getGlobalAxisRadius()) * hitboxScale.x + fwd * getGlobalAxisLength() * hitboxScale.y * 0.6f;
+        max =
+            (t1 * getGlobalAxisRadius() + t2 * getGlobalAxisRadius()) * hitboxScale.x + fwd * getGlobalAxisLength() * hitboxScale.y;
+      }
+
+      // The selection box of the axis handle remains unchanged during object scaling.
+      handle.selectionBoxTransform = linalg::identity;
+
+      inputContext.updateHandle(handle);
+    }
+  }
+
+  size_t getHandleIndex(Handle &inHandle) { return size_t(inHandle.userData); }
+
+  float3 getAxisDirection(size_t index, float4x4 transform) {
+    float3 base{};
+    base[index] = 1.0f;
+    return linalg::normalize(linalg::mul(transform, float4(base, 0)).xyz());
+  }
+
+  /*
+    The centred cube is implemented as a handle with a cube-shaped selection box. 
+    Its behavior draws inspiration from the scaling gizmo in Unity, where the direction of dragging determines how the object scales.
+    
+    Two different solutions for the centred cube were considered. 
+    The first solution involved intersecting the view ray with a plane parallel to the viewport and passing through the object's center. 
+    The second solution relied on the change in cursor position.
+
+    The second solution was chosen for the following reasons:
+      - The first solution proved more challenging to implement since the ray casting depended on the camera direction, which was not available in the input context.
+      - The second solution offers a more intuitive reasoning as the direction of dragging is directly reflected in the cursor position changes.
+  */
+  virtual void grabbed(InputContext &context, Handle &handle) {
+    handle.grabbed = true;
+    dragStartTransform = transform;
+
+    size_t index = getHandleIndex(handle);
+    SPDLOG_DEBUG("Handle {} ({}) grabbed", index, getAxisDirection(index, dragStartTransform));
```

Comment:
since you finished debugging I assume , can be trace or even removed? (likewise for other gizmos)